app/helpers/compute_resources_vms_helper.rb

```diff
@@ -142,8 +142,12 @@ def show_vm_name?
     controller_name != 'hosts' && controller_name != 'compute_attributes'
   end
 
-  def new_host?(host)
-    host.try(:new_record?)
+  def new_host?(host, options = {})
+    if host.nil? && options[:nil_is_new]
```

Comment:
I can see that they do act differently.. i.e. ovirt/vmware partials check `@host` and libvirt/openstack partials check `f.object` (a Fog server object), but I don't understand why we don't change that, rather than complicating this method and accepting the inconsistency.

Unless I'm going mad, I don't really see a reason to initialise `@host` at all in the controller, potentially saving us DB lookups.  The only use I see for it is in the EC2 partial which again checks `@host.nil? || @host.ip.empty?` which could probably be replaced by `new_host?`, another win.

So perhaps just kill `@host` with fire throughout?